### PR TITLE
chore: unify network & chain id

### LINKED ISSUE #1812: Leave only one of network_id and chain_id
Right now we have network_id in network config and chain_id in chain config. But it should be the same thing, or it should be totally separated if not. 
Right now we [have this](https://github.com/Taraxa-project/taraxa-node/blob/develop/libraries/config/src/config.cpp#L257), but we can have situation when network_id will be changed in the config, but not chain id and it could mess something.

### PR DESCRIPTION
**Pull request by MatusKysel**
## Purpose
Fixes #1812

### CODE REVIEW

File: libraries/cli/include/cli/config_jsons/default.json [hunk 1]
```diff
@@ -143,7 +144,6 @@
     ]
   },
   "chain_config": {
-    "chain_id": "0x4",
```
Comment (JakubFornadel) [inline libraries/cli/include/cli/config_jsons/default.json:146]: hmm chain_id sounds like a chain_config param ?

**Review by JakubFornadel — APPROVED** [review]

File: libraries/cli/include/cli/config_jsons/default.json (hunk 1 above)
Comment (MatusKysel) [reply libraries/cli/include/cli/config_jsons/default.json:146]: as we unified network_id and chain_id it can not be chain config param anymore

Comment (MatusKysel) [reply libraries/cli/include/cli/config_jsons/default.json:146]: and one more thing chain config is only optional

**Comment by justinsnapp** [comment]
Does having these be split and not unified serve the purpose of being able to more easily fork Taraxa.  So that you could have taraxafork mainnet and testnet and devnet etc.

**Comment by MatusKysel** [comment]
> Does having these be split and not unified serve the purpose of being able to more easily fork Taraxa. So that you could have taraxafork mainnet and testnet and devnet etc.

using 2 different values is from eth, where network_id is for p2p and chain_id for signing trxs (so they can not use trxs from testnet on mainnet) more here:  https://besu.hyperledger.org/en/stable/Concepts/NetworkID-And-ChainID/ . I do not think there will be scenario where we change chain_id or network_id on real network, but if we planning to have this feature we should not unify them

**Review by mfrankovi — APPROVED** [review]

**Pull request merged**